test/es-module/test-esm-loader-modulemap.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+// Flags: --expose-internals
+
+// This test ensures that the type checking of ModuleMap throws errors appropriately
+
+const assert = require('assert');
+const { URL } = require('url');
+
+const Loader = require('internal/loader/Loader');
+const ModuleMap = require('internal/loader/ModuleMap');
+const ModuleJob = require('internal/loader/ModuleJob');
+const { createDynamicModule } = require('internal/loader/ModuleWrap');
+
+const stubModuleUrl = new URL('file://tmp/test');
+const stubModule = createDynamicModule(['default'], stubModuleUrl);
+const loader = new Loader();
+const moduleMap = new ModuleMap();
+const moduleJob = new ModuleJob(loader, stubModule.module);
+
+assert.throws(() => {
+  moduleMap.get(1);
+}, /TypeError \[ERR_INVALID_ARG_TYPE\]: The "url" argument must be of type string/);
```

Comment:
These should use...

```
common.expectsError(
  () => moduleMap.get(1),
  {
    code: 'ERR_INVALID_ARG_TYPE',
    type: TypeError,
    message: 'The "url" argument must be of type string'
  }
);
```

Reply:
Hi @jasnell I've update the tests to use common.expectsError as suggested

Thanks!